faiss/IndexIVFPQFastScan.cpp

```diff
@@ -286,9 +286,28 @@ void IndexIVFPQFastScan::compute_LUT(
     }
 }
 
-void IndexIVFPQFastScan::sa_decode(idx_t n, const uint8_t* bytes, float* x)
+void IndexIVFPQFastScan::sa_decode(idx_t n, const uint8_t* codes, float* x)
         const {
-    pq.decode(bytes, x, n);
+    size_t coarse_size = coarse_code_size();
+
+#pragma omp parallel
```

Comment:
`parallel if (n > 1)`
or, even better, is n is greater than a certain threshold
please check other `sa_decode` implementations for the appropriate threshold

Reply:
Hi @alexanderguzhva, thanks for the suggestion. I checked other sa_decode. They also don't have the condition. Try to understand it more. `#pragma omp parallel if (n > 1)` means we only parallel execution when n is greater than 1. I'm wondering how we should choose the threshold in general. Do we have any benchmark to run or other methods?

Reply:
no benchmarks, this is a magic constant. Typically, things like 100 or 1000 are used.
https://github.com/facebookresearch/faiss/blob/0c96b0d7e0e399458e7fb0703015f03dbecb614d/faiss/IndexScalarQuantizer.cpp#L186